sqlmesh/core/context.py

```diff
@@ -236,6 +237,11 @@ def __init__(
         self._jinja_macros = JinjaMacroRegistry()
 
         self.path, self.config = t.cast(t.Tuple[Path, Config], next(iter(self.configs.items())))
+        if self.config.model_defaults.dialect is None:
```

Comment:
this should just be a valiadotr inside of model_defaults

Reply:
That's what i originally had, but it errored because `Config.parse_obj()` is called on an incomplete Config dict in [core.config.loader load_config_from_env()](https://github.com/TobikoData/sqlmesh/blob/168fc6b06403277cb7895198dec711ccc3741850/sqlmesh/core/config/loader.py#L108).

No one `Config` must have a default dialect, so my logic was that the validation had to occur after all of them had been collected and we could see the entire set to determine whether any of them had one.

Am I thinking about this wrong?

Reply:
@vchan @izeigerman what do you think?

Reply:
This doesn't check if any `Config` has a default dialect, it only checks the first config of `self.configs` has a default dialect, right?

Reply:
@vchan the goal is just to make model.default_dialect non null

Reply:
> This doesn't check if any `Config` has a default dialect, it only checks the first config of `self.configs` has a default dialect, right?

Good point

Reply:
I agree with @tobymao and this should be a pydantic validator. Maybe `core.config.loader.load_config_from_env` shouldn't return a `Config` object since env vars will probably only supplement/override config fields.

Reply:
Looks like #1136 addressed the incomplete Config issue.